packages/webview_flutter/ios/Classes/JavaScriptChannelHandler.m

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "JavaScriptChannelHandler.h"
+
+@implementation FLTJavaScriptChannel {
+  FlutterMethodChannel* _methodChannel;
+  NSString* _javaScriptChannelName;
+}
+
+- (instancetype)initWithMethodChannel:(FlutterMethodChannel*)methodChannel
+                javaScriptChannelName:(NSString*)javaScriptChannelName {
+  self = [super init];
+  NSAssert(methodChannel != nil, @"methodChannel must not be null.");
+  NSAssert(javaScriptChannelName != nil, @"javaScriptChannelName must not be null.");
+  if (self) {
+    _methodChannel = methodChannel;
+    _javaScriptChannelName = javaScriptChannelName;
+  }
+  return self;
+}
+
+- (void)userContentController:(WKUserContentController*)userContentController
+      didReceiveScriptMessage:(WKScriptMessage*)message {
+  NSAssert(_methodChannel != nil, @"Can't send a message to an unitialized JavaScript channel.");
+  NSAssert(_javaScriptChannelName != nil,
+           @"Can't send a message to an unitialized JavaScript channel.");
+  NSDictionary* arguments = @{
+    @"channel" : _javaScriptChannelName,
+    @"message" : [NSString stringWithFormat:@"%@", message.body]
```

Comment:
if native send json string back flutter, it will got a string that hard to serialized like below.

```json
{
    action = do_somethineg;
    data =     {
        key_1 = some_info;
        key_2 = "some_info";
    };
    timestamp = 1570005270866;
    token = "jwt_token";
}
```

is it possible to add code to serialize NSDictionary and  before send message.body:

```objective-c
- (NSString *)messageFromBody:(NSString *)body {
    NSString *message;
    if ([body isKindOfClass:[NSDictionary class]] ||
        [body isKindOfClass:[NSArray class]]) {
        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:body
                                                           options:NSJSONWritingPrettyPrinted
                                                             error:nil];
        message = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
    }
    if (message.length <= 0) {
        message = [NSString stringWithFormat:@"%@", body];
    }

    return message;
}

...

NSDictionary* arguments = @{
    @"channel" : _javaScriptChannelName,
    @"message" : [self messageFromBody: message.body]
  };
```

Thanks!

Reply:
Do you mind filing this as an issue?

Reply:
Ya, i'm created [HERE](https://github.com/flutter/flutter/issues/42986)